sceval_frontend/src/containers/SensorModule.tsx

```diff
@@ -271,38 +268,21 @@ export const SensorModule: React.FC<SensorModuleProps> = (props: SensorModulePro
         const sensorModule = sessionStorage.getItem('selectedModule');
         const device = sessionStorage.getItem('selectedGateway');
         if (device && sensorModule) {
-            const params = {
-                value: {
-                    name: sensorModuleName,
-                    gatewayID: device, 
-                    id: sensorModule,
-                    interval: 30,
-                    modelId: '0101',
-                    sensing: 'on',
-                    sensors: filteredActiveSensors
-                }
-            };
-            // // update KV store for device + module
-            const deviceURL = MODE_API_BASE_URL + 'devices/' + device + '/kv/sensorModule' + sensorModule;
-            modeAPI.request('PUT', deviceURL, params)
+           // copy the current selected sensor module object and replace the module's name and list of sensors
+            const updatedSensorModuleObj: SensorModuleInterface = Object.assign({}, selectedSensorModuleObj);
+            if (sensorModuleName) {
```

Comment:
@bill-mode just out of curiosity, what is the difference between `Object.assign` and declaring a new variable without `Object.assign`?

Reply:
@mattthewong, Object.assign will create a new Object and also copy the properties from selectedSensorModuleObj. I need to copy all the properties in selectedSensorModuleObj and save them.